presto-delta/src/main/java/com/facebook/presto/delta/DeltaMetadata.java

```diff
@@ -0,0 +1,298 @@
+                session,
+                schemaTableName,
+                tableLocation,
+                deltaTableName.getSnapshotId(),
+                deltaTableName.getTimestampMillisUtc());
+        if (table.isPresent()) {
+            return new DeltaTableHandle(connectorId, table.get(), TupleDomain.all(), Optional.empty());
+        }
+        return null;
+    }
+
+    @Override
+    public List<ConnectorTableLayoutResult> getTableLayouts(
+            ConnectorSession session,
+            ConnectorTableHandle table,
+            Constraint<ColumnHandle> constraint,
+            Optional<Set<ColumnHandle>> desiredColumns)
+    {
+        DeltaTableHandle tableHandle = (DeltaTableHandle) table;
+        if (!config.isFilterPushdownEnabled()) {
+            ConnectorTableLayout layout = new ConnectorTableLayout(new DeltaTableLayoutHandle(tableHandle));
+            return ImmutableList.of(new ConnectorTableLayoutResult(layout, constraint.getSummary()));
+        }
+
+        // Split the predicate into partition column predicate and other column predicates
+        // Only the partition column predicate is fully enforced. Other predicate is partially enforced (best effort).
+        List<TupleDomain<ColumnHandle>> predicate = splitPredicate(constraint.getSummary());
+        TupleDomain<ColumnHandle> enforcedPredicate = predicate.get(0);
+        TupleDomain<ColumnHandle> unenforcedPredicate = predicate.get(1);
+
+        DeltaTableHandle newDeltaTableHandle = tableHandle.withPredicate(
+                constraint.getSummary().transform(DeltaColumnHandle.class::cast),
+                constraint.getSummary().toString(session.getSqlFunctionProperties()));
+
+        ConnectorTableLayout newLayout = new ConnectorTableLayout(
+                new DeltaTableLayoutHandle(newDeltaTableHandle),
+                Optional.empty(),
+                constraint.getSummary(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                ImmutableList.of(),
+                Optional.empty());
+
+        return ImmutableList.of(new ConnectorTableLayoutResult(newLayout, unenforcedPredicate));
+    }
+
+    @Override
+    public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTableLayoutHandle handle)
+    {
+        return new ConnectorTableLayout(handle);
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata(ConnectorSession session, ConnectorTableHandle table)
+    {
+        DeltaTableHandle deltaTableHandle = (DeltaTableHandle) table;
+        checkConnectorId(deltaTableHandle);
+        return getTableMetadata(session, deltaTableHandle.toSchemaTableName());
+    }
+
+    @Override
+    public List<SchemaTableName> listTables(ConnectorSession session, String schemaNameOrNull)
+    {
+        List<String> schemaNames = schemaNameOrNull != null ? ImmutableList.of(schemaNameOrNull) : listSchemaNames(session);
```

Comment:
NIT: We can write it like `schemaNameOrNull == null ? listSchemaNames(session) : ImmutableList.of(schemaNameOrNull)` to improve readability. Just a suggestion.